File: vipclaw-admin/src/main/java/com/vipamp/vipclaw/admin/dto/SysJobResponse.java
package com.vipamp.vipclaw.admin.dto;

import com.vipamp.vipclaw.admin.entity.SysJob;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 定时任务响应 DTO
 *
 * @author vipamp
 * @since 2026-03-16
 */
@Data
@Schema(description = "定时任务响应对象")
public class SysJobResponse {

    /**
     * 任务ID
     */
    @Schema(description = "任务ID", example = "1")
    private Long id;

    /**
     * 任务名称
     */
    @Schema(description = "任务名称", example = "示例任务")
    private String jobName;

    /**
     * 任务组名
     */
    @Schema(description = "任务组名", example = "DEFAULT")
    private String jobGroup;

    /**
     * 执行类全路径
     */
    @Schema(description = "执行类全路径", example = "com.vipclaw.admin.job.SampleJob")
    private String jobClass;

    /**
     * Cron执行表达式
     */
    @Schema(description = "Cron执行表达式", example = "0/5 * * * * ?")
    private String cronExpression;

    /**
     * 状态（0-暂停，1-运行）
     */
    @Schema(description = "状态（0-暂停，1-运行）", example = "1")
    private Integer jobStatus;

    /**
     * 是否允许并发（0-禁止，1-允许）
     */
    @Schema(description = "是否允许并发（0-禁止，1-允许）", example = "1")
    private Integer concurrent;

    /**
     * 任务描述
     */
    @Schema(description = "任务描述", example = "这是一个示例定时任务")
    private String description;

    /**
     * 创建时间
     */
    @Schema(description = "创建时间", example = "2026-03-16 12:00:00")
    private LocalDateTime createTime;

    /**
     * 更新时间
     */
    @Schema(description = "更新时间", example = "2026-03-16 12:00:00")
    private LocalDateTime updateTime;

    /**
     * 从实体对象转换
     *
     * @param job 定时任务实体
     * @return 定时任务响应对象
     */
    public static SysJobResponse fromEntity(SysJob job) {
        if (job == null) {
            return null;
        }
        SysJobResponse response = new SysJobResponse();
        response.setId(job.getId());
        response.setJobName(job.getJobName());
        response.setJobGroup(job.getJobGroup());
        response.setJobClass(job.getJobClass());
        response.setCronExpression(job.getCronExpression());
        response.setJobStatus(job.getJobStatus());
        response.setConcurrent(job.getConcurrent());
        response.setDescription(job.getDescription());
        response.setCreateTime(job.getCreateTime());
        response.setUpdateTime(job.getUpdateTime());
        return response;
    }
}

File: vipclaw-admin/src/main/kotlin/com/vipamp/vipclaw/admin/dto/SysJobResponse.kt
package com.vipamp.vipclaw.admin.dto

import com.vipamp.vipclaw.admin.entity.SysJob
import io.swagger.v3.oas.annotations.media.Schema
import java.time.LocalDateTime

/**
 * Scheduled job response object
 */
@Schema(description = "Scheduled job response object")
data class SysJobResponse(
    @Schema(description = "Job ID", example = "1")
    val id: Long? = null,
    @Schema(description = "Job name", example = "Sample Job")
    val jobName: String? = null,
    @Schema(description = "Job group name", example = "DEFAULT")
    val jobGroup: String? = null,
    @Schema(description = "Full path of execution class", example = "com.vipclaw.admin.job.SampleJob")
    val jobClass: String? = null,
    @Schema(description = "Cron expression", example = "0/5 * * * * ?")
    val cronExpression: String? = null,
    @Schema(description = "Status (0:paused, 1:running)", example = "1")
    val jobStatus: Int? = null,
    @Schema(description = "Allow concurrent execution (0:no, 1:yes)", example = "1")
    val concurrent: Int? = null,
    @Schema(description = "Job description", example = "This is a sample scheduled job")
    val description: String? = null,
    @Schema(description = "Whether public (0:no, 1:yes)", example = "1")
    val isPublic: Int? = null,
    @Schema(description = "Creator", example = "admin")
    val creator: String? = null,
    @Schema(description = "Creation time", example = "2026-03-16 12:00:00")
    val createTime: LocalDateTime? = null,
    @Schema(description = "Update time", example = "2026-03-16 12:00:00")
    val updateTime: LocalDateTime? = null,
) {
    companion object {
        @JvmStatic
        fun fromEntity(entity: SysJob?): SysJobResponse {
            if (entity == null) return SysJobResponse()
            return SysJobResponse(
                id = entity.id,
                jobName = entity.jobName,
                jobGroup = entity.jobGroup,
                jobClass = entity.jobClass,
                cronExpression = entity.cronExpression,
                jobStatus = entity.jobStatus,
                concurrent = entity.concurrent,
                description = entity.description,
                isPublic = entity.isPublic,
                creator = entity.creator,
                createTime = entity.createTime,
                updateTime = entity.updateTime,
            )
        }
    }
}
